### ISSUE TITLE
y_groups

### ISSUE THREAD

**Issue by AndiChs** [pawn-lang/YSI-Includes#139]
If you try to remove a player from a group while you loop through it the server will enter in an endless loop.

**Comment by Crayder** [comment]
Like, a loop with iterators?

Show us your code.

**Comment by AndiChs** [comment]
Yes, just like that. 
foreach(new i : Group_X) 
    if(condition)
       Group_SetPlayer(Group_X, i, false)

**Comment by Crayder** [comment]
Yeah, that's your problem then.

You'll need to make a safe removing alternative. 

```
foreach(new i : Group_X)
	if(condition)
		Iter_SafeRemove(Group_X, i, i);
		Group_SetPlayer(Group_X, i, false);
```

**Comment by Y-Less** [comment]
Not quite, because `Iter_SafeRemove` takes a reference parameter to return the NEXT loop item, so that code will remove them from the iterator, return the next iteration, then remove them from that next group:

```pawn
foreach (new i : Group_X)
{
	if (condition)
	{
		new next;
		Iter_SafeRemove(Group_X, i, next);
		Group_SetPlayer(Group_X, i, false);
		i = next;
	}
}
```

However, there are in-built group special iterators:

```pawn
foreach (new i : GroupPlayer(X))
{
	Group_SetPlayer(X, i, false);
}
```

But to be fair, I don't know how they react to modification mid loop either (which has never been a good idea in any language).
